tests: improved the unit test code coverage to SlideUpViewPresenter.swift (SDKCF-6383)

Esakkiraja-Pothikannan opened the pull request:
# Description
- Added `slide-up-trigger-without-messageBody.json` file with an invalid response
- Added coverage to `guard else` statement in `viewDidInitialize()`
- Removed unused `completionHandler` 

## Links
- SDKCF-6383

# Checklist
- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md)
- [x] I have added to the [changelog](../blob/master/CHANGELOG.md#Unreleased)
- [x] I wrote/updated tests for new/changed code
- [ ] I removed all sensitive data **before every commit**, including API endpoints and keys, and internal links
- [x] I ran `fastlane ci` without errors
- [x] All project file changes are replicated in `SampleSPM/SampleSPM.xcodeproj` project

Esakkiraja-Pothikannan committed:
test: improved the unit test code coverage to SlideUpViewPresenter.swift file

ts-robih-robihamanto commented:
@imaginaris 
Do we need to add a new file in this case `slide-up-trigger-without-messageBody.json` to `SampleSPM/SampleSPM.xcodeproj`? or it's just for the Spec file?

Esakkiraja-Pothikannan committed:
tests: reverted the changes (SDKCF-6383)

Esakkiraja-Pothikannan committed:
tests: fixed indentation error (SDKCF-6383)

Esakkiraja-Pothikannan commented:
> @imaginaris Do we need to add a new file in this case `slide-up-trigger-without-messageBody.json` to `SampleSPM/SampleSPM.xcodeproj`? or it's just for the Spec file?

This is only for `UITests` target.

imaginaris commented on CHANGELOG.md:
```diff
@@ -6,9 +6,11 @@
-	- Added a new file for EventType unit test to increase code coverage [SDKCF-6378]
-	- Added more unit test to increase code coverage [SDKCF-6379]
-	- Added testing code to cover all code in the UserInfoProvider [SDKCF-6380]
+	- Improved unit test code coverage to
```
👍

imaginaris commented:
> @imaginaris
> Do we need to add a new file in this case `slide-up-trigger-without-messageBody.json` to `SampleSPM/SampleSPM.xcodeproj`? or it's just for the Spec file?

That's right. All new files, including JSON files, should be also added to `SampleSPM.xcodeproj`

Esakkiraja-Pothikannan commented on RInAppMessaging.xcodeproj/project.pbxproj:
```diff
@@ -73,6 +73,7 @@
 		45BDFD032421D311004DEA0C /* CampaignsListManagerSpec.swift in Sources */ = {isa = PBXBuildFile; fileRef = 45BDFD022421D311004DEA0C /* CampaignsListManagerSpec.swift */; };
 		45BDFD052421EE7B004DEA0C /* SharedMocks.swift in Sources */ = {isa = PBXBuildFile; fileRef = 45BDFD042421EE7B004DEA0C /* SharedMocks.swift */; };
 		45BDFD0724232714004DEA0C /* PublicAPISpec.swift in Sources */ = {isa = PBXBuildFile; fileRef = 45BDFD0624232714004DEA0C /* PublicAPISpec.swift */; };
+		45D064D029D2ECE10009DFB2 /* slide-up-trigger-without-messageBody.json in Resources */ = {isa = PBXBuildFile; fileRef = 45D064CE29D2ECE10009DFB2 /* slide-up-trigger-without-messageBody.json */; };
```
@imaginaris Why test target file should be added to `SampleSPM.xcodeproj` ?

imaginaris replied:
Because we use `SampleSPM.xcodeproj` to run tests in SPM mode. The CI will attempt to build 'Tests' target in `SampleSPM.xcodeproj`. So it's better to keep that project in sync with the original RInAppMessaging.xcodeproj

malayli commented on Tests/UITests/SlideUpViewSpec.swift:
```diff
@@ -40,6 +40,20 @@ class SlideUpViewSpec: QuickSpec {
+
+                beforeEach {
+                    launchAppIfNecessary(context: "slide-up-trigger-without-messageBody")
+                    if !iamView.exists {
```
@Esakkiraja-Pothikannan Why is this condition needed?

imaginaris replied:
Yeah, In this case instead of this condition you should check if iamView is not presented: `expect(iamView.exists).to(beFalse())`

Esakkiraja-Pothikannan replied:
I don't think, it makes any difference from the current implementation. so it's fine with `if condition` right ?

imaginaris replied:
This condition just does not match the expected behaviour. In other contexts it ensures the iamView is present before running any tests cases. In this context you don't want that to happen. So just having `app.buttons["purchase_successful"].tap()` should do

Esakkiraja-Pothikannan replied:
ok got it, I understood other contexts `if condition` in a different way. So `expect(iamView.exists).to(beFalse())` and if `!iamView.exists {` is not required, we can directly call `app.buttons["purchase_successful"].tap()`. Will update this.

imaginaris commented on Tests/UITests/SlideUpViewSpec.swift:
```diff
@@ -40,6 +40,20 @@ class SlideUpViewSpec: QuickSpec {
+                }
+
+                it("should not construct the slideup view") {
+                    expect(iamView.exists).to(beFalse())
```
It might take some time for IAM decide if `iamView` should be displayed or not. So you need to use `.toAfterTimeout(beFalse(), timeout: 2)` predicate.

Esakkiraja-Pothikannan committed:
test: add 'slide-up-trigger-without-messageBody.json' to SampleSPM project (SDKCF-6383)

Esakkiraja-Pothikannan committed:
test: updated some review comments (SDKCF-6383)

ts-robih-robihamanto committed:
tests: Add testing code to cover all code in the EventMatcher (SDKCF-6390)

* tests: update unit test to increase code coverage in EventMatcher (SDKCF-6390)

Updated to 100% coverage

* chore: update changelog to include EventMatcher update unit test (SDKCF-6390)

* tests: update unit test case description (SDKCF-6390)

* chore: update the testing code to match with the description to avoid confusion (SDKCF-6390)

ts-robih-robihamanto committed:
tests: improve Code Coverage of ConfigurationService (SDKCF-6388)

* tests: Add testing code to increase coverage in ConfigurationService (SDKCF-6388)

* chore: Update changelog to include current working ticket (SDKCF-6388)

* tests: Add testing to cover more case  (SDKCF-6388)

Added for assertionFailure in building buildRequestHeader() and buildHttpBody()

malayli commented on Sources/RInAppMessaging/EventMatcher.swift:
```diff
@@ -80,11 +80,10 @@ internal class EventMatcher: EventMatcherType {
-        if campaign.isTooltip {
-            return allTriggersSatisfied && events.contains(where: { $0.type == .viewAppeared })
-        } else {
+        guard campaign.isTooltip else {
```
@Esakkiraja-Pothikannan Was `guard` needed for coverage here?
Or you made this change because it is more readable than if else?

Esakkiraja-Pothikannan commented:
closing this as [PR-266](https://github.com/rakutentech/ios-inappmessaging/pull/266) has all the changes
